libs/ui-lib/lib/ocm/components/clusterConfiguration/operators/FenceAgentsRemediationCheckbox.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import { FormGroup, HelperText, HelperTextItem, Tooltip } from '@patternfly/react-core';
+import { getFieldId, PopoverIcon, FENCE_AGENTS_REMEDIATION_LINK } from '../../../../common';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons/dist/js/icons/external-link-alt-icon';
+import { OcmCheckboxField } from '../../ui/OcmFormFields';
+import { useNewFeatureSupportLevel } from '../../../../common/components/newFeatureSupportLevels';
+import NewFeatureSupportLevelBadge from '../../../../common/components/newFeatureSupportLevels/NewFeatureSupportLevelBadge';
+import { SupportLevel } from '@openshift-assisted/types/./assisted-installer-service';
+
+const FENCE_AGENTS_REMEDIATION_FIELD_NAME = 'useFenceAgentsRemediation';
+const FENCE_AGENTS_REMEDIATION_FEATURE_ID = 'FENCE_AGENTS_REMEDIATION';
+
+const FenceAgentsRemediationLabel = ({
+  disabledReason,
+  supportLevel,
+}: {
+  disabledReason?: string;
+  supportLevel?: SupportLevel;
+}) => {
+  return (
+    <>
+      <Tooltip hidden={!disabledReason} content={disabledReason}>
+        <span>Install Fence Agents Remediation </span>
+      </Tooltip>
+      <PopoverIcon
+        id={FENCE_AGENTS_REMEDIATION_FEATURE_ID}
+        component={'a'}
+        bodyContent={'No additional requirements needed'}
+      />
+      <NewFeatureSupportLevelBadge
+        featureId={FENCE_AGENTS_REMEDIATION_FEATURE_ID}
+        supportLevel={supportLevel}
+      />
+    </>
+  );
+};
+
+const FenceAgentsRemediationHelperText = () => {
+  return (
+    <HelperText>
+      <HelperTextItem variant="indeterminate">
+        Externally fences failed nodes using power controllers.{' '}
+        <a href={FENCE_AGENTS_REMEDIATION_LINK} target="_blank" rel="noopener noreferrer">
+          {'Learn more'} <ExternalLinkAltIcon />
+        </a>
+      </HelperTextItem>
+    </HelperText>
+  );
+};
+
+const FenceAgentsRemediationCheckbox = ({ disabledReason }: { disabledReason?: string }) => {
+  const fieldId = getFieldId(FENCE_AGENTS_REMEDIATION_FIELD_NAME, 'input');
+  const featureSupportLevel = useNewFeatureSupportLevel();
+  const [disabledReasonFenceAgentsRemediation, setDisabledReason] = useState<string | undefined>();
+
+  React.useEffect(() => {
+    const reason = featureSupportLevel.getFeatureDisabledReason(
+      FENCE_AGENTS_REMEDIATION_FEATURE_ID,
+    );
+    setDisabledReason(reason);
+  }, [featureSupportLevel]);
```

Comment:
why are you using `useState` + `useEffect` ? This can be done without it.

Applies for all others as well.

Reply:
I was coping https://github.com/openshift-assisted/assisted-installer-ui/blob/edf64c4aac1ae424d478f95223b1520f69d5c1b6/libs/ui-lib/lib/ocm/components/clusterConfiguration/operators/KmmCheckbox.tsx#L46. LMK if you think I should change it.

Reply:
Yes, please lets change it. No need to repeat a bad pattern.

Reply:
We are passing this value as a prop - https://github.com/openshift-assisted/assisted-installer-ui/blob/7335061627634f25c65438b22ef0c8d3f50d8a44/libs/ui-lib/lib/ocm/components/clusterWizard/OperatorsStep.tsx#L451, so I am not sure if we need to. We get it here - https://github.com/openshift-assisted/assisted-installer-ui/blob/7335061627634f25c65438b22ef0c8d3f50d8a44/libs/ui-lib/lib/ocm/components/clusterWizard/OperatorsStep.tsx#L240

Reply:
WDYT ? @rawagner

Reply:
Ok, I see. Great that you removed this duplication!